operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ResourceConfiguration.java

```diff
@@ -67,7 +68,12 @@ static boolean currentNamespaceWatched(Set<String> namespaces) {
         throw new IllegalStateException(
             "Parent ConfigurationService must be set before calling this method");
       }
-      targetNamespaces = Collections.singleton(parent.getClientConfiguration().getNamespace());
+      String namespace = parent.getClientConfiguration().getNamespace();
```

Comment:
I'm not sure about this change, is possibly going to break user's (especially CIs) environment when a provider doesn't set the default namespace.

I would suggest that if the namespace is `null` you can output a warning message and automatically set it to `default`.

Reply:
Btw not sure if this can happen in a real environment (I mean pod), probably just locally, for that purpose might be better to have an exception, so developer won't search for the problem unnecessarily.

Reply:
A possibility, for the deployment, is to use a [downwardAPI](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#the-downward-api).

Reply:
I mean, a default namespace will be always there.

Reply:
Yes, this should only happen when outside of the cluster…

Reply:
> not sure if this can happen in a real environment

I'm mostly worried about CI ... but, well, an additional command doesn't hurt as well 🤷

Reply:
Let's see how it goes and maybe revisit if needed?